hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/has/isauthorizedraw/IsAuthorizedRawCall.java

```diff
@@ -0,0 +1,266 @@
+        if (!isValidAccount(accountNum, signatureType)) return bail.apply(INVALID_ACCOUNT_ID);
+        final var account = requireNonNull(enhancement.nativeOperations().getAccount(accountNum));
+
+        // If ED then require a key on the account
+        final Optional<Key> key;
+        if (signatureType == SignatureType.ED) {
+            key = Optional.ofNullable(account.key());
+            if (key.isEmpty()) return bail.apply(FAIL_INVALID /* should be: "account must have key" */);
+        } else key = Optional.empty();
+
+        // Key must be simple (for isAuthorizedRaw)
+        if (key.isPresent()) {
+            final Key ky = key.get();
+            final boolean keyIsSimple = !ky.hasKeyList() && !ky.hasThresholdKey();
+            if (!keyIsSimple) return bail.apply(FAIL_INVALID /* should be: "account key must be simple" */);
+        }
+
+        // Key must match signature type
+        if (key.isPresent()) {
+            if (!switch (signatureType) {
+                case ED -> key.get().hasEd25519();
+                case EC -> false;
+                default -> false;
+            }) return bail.apply(INVALID_SIGNATURE_TYPE_MISMATCHING_KEY);
+        }
+
+        // Finally: Do the signature validation we came here for!
+        final boolean authorized =
+                switch (signatureType) {
+                    case EC -> validateEcSignature(address, frame);
+                    case ED -> validateEdSignature(account, key.get());
+                    default -> false;
+                };
+
+        final var result = gasOnly(
+                successResult(encodedAuthorizationOutput(authorized), HARDCODED_GAS_REQUIREMENT_GAS), SUCCESS, true);
+        return result;
+    }
+
+    /** Validate EVM signature - EC key - via the ECRECOVER precompile */
+    public boolean validateEcSignature(@NonNull final Address address, @NonNull final MessageFrame frame) {
+        requireNonNull(address, "address");
+        requireNonNull(frame, "frame");
+
+        // Call the ECRECOVER precompile directly (meaning, not by executing EVM bytecode, just by
+        // using the class provided by Besu).
+
+        final var input = formatEcrecoverInput(messageHash, signature);
+        if (input.isEmpty()) return false;
+        final var ecrecoverResult = ecPrecompile.computePrecompile(Bytes.wrap(input.get()), frame);
+
+        // ECRECOVER's output always has status SUCCESS but the result Bytes are either empty or the recovered address.
+        if (ecrecoverResult.getOutput().isEmpty()) return false;
+        final var recoveredAddress =
+                ecrecoverResult.getOutput().slice(12).toArray(); // LAST 20 bytes are where the EVM address is
+
+        // ECRECOVER produced an address:  Must match our account's alias address
+        final var givenAddress = explicitFromHeadlong(address);
+        final var addressesMatch = 0 == Arrays.compare(recoveredAddress, givenAddress);
+
+        return addressesMatch;
```

Comment:
```suggestion
        return 0 == Arrays.compare(recoveredAddress, givenAddress);
```

Reply:
If you prefer you could also keep it on separate line, but then I would probably suggest that we could use `boolean` instead of `var`.

Reply:
WIll not change (see comment above).

(Unless you disagree with comment above - let me know.)

Reply:
Good idea, I believe that too.